nexus_constructor/model/component.py

```diff
@@ -462,13 +453,14 @@ def _create_transformation_vectors_for_pixel_offsets(
             z_offsets = self.get_field_value(Z_PIXEL_OFFSET)
         except AttributeError:
             z_offsets = np.zeros_like(x_offsets)
+        if not isinstance(x_offsets, list):
```

Comment:
May be a small comment here as to why flatten is used. Not very clear as of now.

Reply:
It is old code really, once it is saved to a json it is in a list. However, I think internally it will be a np.array when you define the detector from the UI.